### DISCUSSION TITLE
how to configure grafana env

### DISCUSSION THREAD

**Discussion by yangvipguang** [prometheus-operator/kube-prometheus#597] [General]
I want to configure grafana env  。But I have no idea with the format of grafana libjsonnet。Could any one can help me with a example。
![image](https://user-images.githubusercontent.com/11416887/86204945-57779080-bb9b-11ea-9b3d-0f27594e010a.png)

And will  the value of env conflict with the value of config sections。
---

We are still happy to chat with you in the #prometheus-operator channel on Kubernetes Slack!

**Comment by songrijie** [comment]
check [this comment](https://github.com/coreos/kube-prometheus/issues/562#issuecomment-652192186) I made.

**Comment by yangvipguang** [comment]
@songrijie  Thanks a lot. But one question is will the value of env conflict with the value of config sections，and are both of the  equal ?

**Comment by songrijie** [comment]
@yangvipguang  env wins.  

look at [this](https://grafana.com/docs/grafana/latest/administration/configuration/#configure-with-environment-variables)

> All options in the configuration file can be `overridden` using environment variables using the syntax

**Comment by yangvipguang** [comment]
@songrijie   Now i have add the configuration with your suggustions。But when Building with jsonnet ，it occurs some issuess。
![image](https://user-images.githubusercontent.com/11416887/86229870-e6e46a00-bbc2-11ea-988c-266d60233412.png)

ERROR：
![image](https://user-images.githubusercontent.com/11416887/86230642-ec8e7f80-bbc3-11ea-9d17-1f1d038aae98.png)

**Reply by olegmayko** [reply]
you may try to use/import "newer" version of lib `local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';`

**Reply by songrijie** [reply]
try ksonnet v4, and this works for me...  you may define a new variable instead. 

```
local k4 = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
local env = k4.apps.v1.deployment.mixin.spec.template.spec.containersType.envType;
```
